chain/client/src/sync.rs

```diff
@@ -1083,8 +1085,18 @@ impl StateSync {
         highest_height_peers: &Vec<FullPeerInfo>,
         tracking_shards: Vec<ShardId>,
     ) -> Result<StateSyncResult, near_chain::Error> {
+        let prev_hash = chain.get_block_header(&sync_hash)?.prev_hash().clone();
         let now = Utc::now();
-        let (request_block, have_block) = self.sync_block_status(sync_hash, chain, now)?;
+
+        // This is possible if the head is within `STATE_FETCH_HORIZON` from the end of the very
+        // first epoch
+        if prev_hash == CryptoHash::default() {
+            self.state_sync_time.clear();
+            assert!(false); // MOO
```

Comment:
should this be removed or at least move to debug assert

Reply:
Yes, MOO means "review before submitting the PR and remove". The process glitched somewhere.